### PR TITLE
Matched ShootingStarMgr::initSchedule

### PR DESCRIPTION
**Pull request by TheGreatB3**
This change is [Reviewable](https://reviewable.io/reviews/zeldaret/botw/114)

### CODE REVIEW

File: src/KingSystem/World/worldShootingStarMgr.cpp [hunk 1]
```diff
@@ -30,27 +30,32 @@ void ShootingStarMgr::init_(sead::Heap* heap) {
         initSchedule();
 }
 
-// NON_MATCHING
 void ShootingStarMgr::initSchedule() {
-    if (sHours.start >= sHours.end) {
-        sead::FixedObjArray<u32, 24> validHours;
+    const s32& start_hour = sHours.start;
```
Comment (leoetlino) [inline src/KingSystem/World/worldShootingStarMgr.cpp:34]: Does this match without the reference?

Comment (TheGreatB3) [reply src/KingSystem/World/worldShootingStarMgr.cpp:34]: It doesn't. `start_hour` has to be a reference here. But `end_hour` matches with or without the reference.

Comment (TheGreatB3) [reply src/KingSystem/World/worldShootingStarMgr.cpp:34]: But it wouldn't make much sense for `end_hour` to be a reference here, since it's computed.

Comment (leoetlino) [reply src/KingSystem/World/worldShootingStarMgr.cpp:34]: That's unfortunate but not that strange so I think it's fine for start_hour to be a reference.

**Pull request merged**